### PR TITLE
Conventional fxml views

### PR DESCRIPTION
**Pull request by ghazyami**
# Conventional FXML views

Conventional FXML views offers a convenient way to load FXML based views, by following structural and naming conventions a view can easily be used without manually loading the `.fxml` file and associated stylesheet and resource bundle

- Class that extends `io.quarkiverse.fx.FXMLView`
- FXML file with the same name as the class above, located at under the same directory structure as the class above under `src/main/resources`
- Stylesheet file with the same name as the class above, located at under the same directory structure as the class above under `src/main/resources` (Optional)
- Resource bundle file with the same name as the class above, located at under the same directory structure as the class above under `src/main/resources` (Optional)

## Example

```java
// src/main/java/org/example/ContactCardView.java
// src/main/resources/org/example/ContactCardView.fxml
// src/main/resources/org/example/ContactCardView.css (Optional)
// src/main/resources/org/example/ContactCardView.properties (Optional)

// Having these files in place we can simply load the view by injecting the instance then calling getNode() on it

@Inject
ContactCardView view;

public void start(@Observes @PrimaryStage final Stage stage) {
    Scene scene = new Scene(view.getNode());
    stage.setScene(scene);
    stage.show();
}
```

##  ContactCardView.java
```java
@Dependent
public class ContactCardView extends FXMLView<VBox> {}
```
Extending FXMLView specifying the root node as defined on `ContactCardView.fxml`

## ContactCardView.fxml
```xml
<?xml version="1.0" encoding="UTF-8"?>

<?import javafx.geometry.Insets?>
<?import javafx.scene.control.Label?>
<?import javafx.scene.layout.VBox?>

<VBox alignment="CENTER" spacing="20.0" xmlns="http://javafx.com/javafx/21.0.1" xmlns:fx="http://javafx.com/fxml/1"
styleClass="red-bg">
   <padding><Insets bottom="10.0" left="10.0" right="10.0" top="10.0"/></padding>
   <Label id="firstName" text="John"/>
   <Label id="lastName" text="Doe"/>
   <Label id="location" text="%location"/>
</VBox>
```
`styleClass` set to a class defined on `ContactCardView.css`
`text` set to an entry defined on `ContactCardView.properties` resource bundle

## ContactCardView.css
```css
.red-bg {
    -fx-background-color: red;
}
```

##  ContactCardView.properties
```properties
location=Some location
```

### CODE REVIEW

**Comment by CodeSimcoe** [comment]
I kind of doubt this belongs in this extension.
One good hint for that is the deployment processor is not involved.
It could be used as an external library, not as part of this extension.

I leave it open for other feedbacks.

**Comment by ghazyami** [comment]
Good point!, Implementation relies on CDI. It can be an external library that only works with this extension or updated to handle other DI environments.

There are already similar libraries like [afterburner.fx](https://github.com/AdamBien/afterburner.fx) (standalone) and [ignite](https://github.com/gluonhq/ignite) (micronaut).

So I felt it's suitable to have this to integrate with CDI

**Comment by hantsy** [comment]
I would like there is an annotation `@FxmlView` to attach the view to the controller, make the codes are similar to the MVC architecture. 

Personally I have used [JavaFX Weaver](https://github.com/rgielen/javafx-weaver) to integrate CDI/Weld with JavaFX in my example project, https://github.com/hantsy/cargotracker-regapp/blob/master/src/main/java/org/eclipse/cargotrakcer/regapp/ui/HandlingReportController.java#L28

File: runtime/src/main/java/io/quarkiverse/fx/FXMLView.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+import javafx.beans.value.ObservableValue;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.Node;
+import javafx.scene.Scene;
+
+public abstract class FXMLView<T extends Node> {
+
+    private static final Logger LOGGER = Logger.getLogger(FXMLView.class);
+    protected static final String FXML_EXT = ".fxml";
+    protected static final String CSSL_EXT = ".css";
```
Comment (hantsy) [inline runtime/src/main/java/io/quarkiverse/fx/FXMLView.java:23]: CSSL_EXT ->  CSS_EXT

**Review by hantsy — APPROVED** [review]

**Comment by hantsy** [comment]
Looks good, but I think my issue #43 can be another topic, use annotation instead of the view extended class.

I think most logic of the `FXMLView` can be extracted to into a standalone class(like the [fxwear bean](https://github.com/rgielen/javafx-weaver/blob/master/core/src/main/java/net/rgielen/fxweaver/core/FxWeaver.java)) and reused for scan views by annotations.

**Pull request closed without merging**